NewsApp/NewsApp/Controller/NewsController.swift

```diff
@@ -9,38 +9,43 @@
 import UIKit
 
 class NewsController: UITableViewController {
-    private let apiClient = APIClient()
-    private var newsArticles: [Article] = []
+    private let newsService: NewsService = NewsServiceImpl()
+    var category: Category?
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        apiClient.send(GetTopHeadlines()) { [weak self] result in
-            switch result {
-            case .success(let response):
-                let articles = response.articles ?? []
-                self?.newsArticles = articles
-                DispatchQueue.main.async {
-                    self?.tableView.reloadData()
-                }
-            case .failure(let error):
-                debugPrint(error.localizedDescription)
+        self.refreshControl?.addTarget(self, action: #selector(reloadDataTableView), for: UIControl.Event.valueChanged)
+        self.title = category?.title
+        self.newsService.getTopHeadlines(for: category) {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    @objc func reloadDataTableView() {
+        self.newsService.getTopHeadlines(for: category) {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
             }
         }
+        self.refreshControl?.endRefreshing()
     }
```

Comment:
On a une duplication de code, est-ce que c'est pas possible de factoriser avec plus haut ?

Reply:
fix in "fix NewsController"